src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -36,9 +48,29 @@ private Event createEventFromEntity(Entity entity) {
     @Override 
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
         Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
-        
+
         if (request.getParameterMap().containsKey("school_id")) {
             Long school_id = Long.parseLong(request.getParameter("school_id"));
+
+            String[] event_types_arr = {};
+            if (request.getParameterMap().containsKey("event_type")) {
+                event_types_arr = request.getParameterValues("event_type");
+            }
+            else {
+                event_types_arr = Event.getEventTypes();
+            }
+            ArrayList<String> event_types = new ArrayList<>(Arrays.asList(event_types_arr));
+
+            String[] subjects_arr = {};
+            if (request.getParameterMap().containsKey("subject")) {
+                subjects_arr = request.getParameterValues("subject");
+            }
+            else {
+                subjects_arr = Event.getSubjects();
```

Comment:
If there are no restrictions on the subject or event type, simply just don't add any additional filters to the query and only query by school id. If you're worried about the subject/event type fields not being apart of the supported types, then you should add a check for this before you submit the item to the database.